### PR TITLE
Fat Pr - multiple bugs resolution

### LINKED ISSUE #64: Check if chanlimit is used in JOIN
Oopsie

### LINKED ISSUE #76: JOIN 0
JOIN 0 should quit all channels

### PR DESCRIPTION
**Pull request by Neffi42**
- Clients now have a joinedChannels count, it can be incremented, decremented and set to 0
- when joining new channels JOIN now checks if the client is over CHANLIMIT
- when quitting a channel, the client's joinedChannels gets decremented
- added chanlimit protection
- added protection against rejoin
- created PART command
- created joinZero
- added deletion of channels when empty
- updated error messages
- delete channel when quiting
- updated WHO command

### CODE REVIEW

File: src/Client.cpp [hunk 1]
```diff
@@ -59,6 +61,18 @@ std::string Client::getModes() const {
 	return modes.empty() ? "" : "+" + modes;
 }
 
+size_t Client::getJoinedChannelsNbr() const {
+	size_t n = 0;
+
+	for (size_t i = 0; i < Server::getInstance().getChannels().size(); ++i) {
+		if (Server::getInstance().getChannels()[i]->isUserOnChannel(
```
Comment (iamantoinev) [inline src/Client.cpp]: Prefere create a local variable to store the array instead
And is not already there a function that returns a list of joined channels ?

**Review by antoineverin — CHANGES_REQUESTED** [review]

File: src/Client.cpp (hunk 1 above)
Comment (Neffi42) [reply src/Client.cpp]: There wasn't any methods but now there is :)

File: src/Client.cpp [hunk 2]
```diff
@@ -59,6 +62,18 @@ std::string Client::getModes() const {
 	return modes.empty() ? "" : "+" + modes;
 }
 
+std::vector<Channel*> Client::getJoinedChannels() const {
+	const std::vector<Channel*> channels = Server::getInstance().getChannels();
```
Comment (iamantoinev) [inline src/Client.cpp]: could be a reference

**Review by antoineverin — CHANGES_REQUESTED** [review]

File: src/Client.cpp (hunk 2 above)
Comment (iamantoinev) [reply src/Client.cpp]: fix in #93

**Review by antoineverin — APPROVED** [review]

**Review by croueAlec — APPROVED** [review]

**Pull request merged**